internal/driver/fetch.go

```diff
@@ -529,16 +529,23 @@ func convertPerfData(perfPath string, ui plugin.UI) (*os.File, error) {
 	ui.Print(fmt.Sprintf(
 		"Converting %s to a profile.proto... (May take a few minutes)",
 		perfPath))
-	profile, err := newTempFile(os.TempDir(), "pprof_", ".pb.gz")
+	// Get a tempFile
+	tempFilename, err := getNextTempFile(os.TempDir(), "pprof_", ".pb.gz")
 	if err != nil {
 		return nil, err
 	}
-	deferDeleteTempFile(profile.Name())
-	cmd := exec.Command("perf_to_profile", perfPath, profile.Name())
+	cmd := exec.Command("perf_to_profile", perfPath, tempFilename)
 	if err := cmd.Run(); err != nil {
-		profile.Close()
 		return nil, fmt.Errorf("failed to convert perf.data file. Try github.com/google/perf_data_converter: %v", err)
 	}
+
+	// Read the profile generated by perf_to_profile
```

Comment:
I'd remove this comment as well. And it's not really reading the profile here, it only opens the file.